fogros2/fogros2/aws.py

```diff
@@ -252,16 +235,31 @@ def __init__(self, ami_image, region="us-west-1", ec2_instance_type="t2.micro",
         self.ec2_instance_disk_size = disk_size  # GB
         self.aws_ami_image = ami_image
 
-        # key & security group names
-        self.ec2_security_group = "FOGROS2_SECURITY_GROUP"
-        self.ec2_key_name = "FogROS2KEY-" + self.unique_name
-        self.ssh_key_path = os.path.join(self.working_dir, self.ec2_key_name + ".pem")
-
         # aws objects
         self.ec2_instance = None
         self.ec2_resource_manager = boto3.resource("ec2", self.region)
         self.ec2_boto3_client = boto3.client("ec2", self.region)
 
+        # Check for name collision among other AWS instances
+        cur_instances = self.ec2_boto3_client.describe_instances(
```

Comment:
Did you explore seeing if you could get AWS to reject a duplicate name?


[Jeff Ichnowski](https://app.slack.com/team/UK66LLFHN)  [15 minutes ago](https://learningsystems.slack.com/archives/C01MZN8ETMH/p1652320759477599?thread_ts=1652320383.335449&cid=C01MZN8ETMH)
thus have it be 1 AWS call to create in the normal case


[Jeff Ichnowski](https://app.slack.com/team/UK66LLFHN)  [14 minutes ago](https://learningsystems.slack.com/archives/C01MZN8ETMH/p1652320768364969?thread_ts=1652320383.335449&cid=C01MZN8ETMH)
and loop on failure in the exception case?


Jeff Ichnowski  14 minutes ago
right now it looks like its doing 2 calls in the normal case--the first call gets a list of instances, which could be a long list (some people have 10000s of instances)

Reply:
According to what I could find, the "name" tag is not used by AWS at all (since it relies on the internal instance IDs), so I don't think there is a good way to autoreject the creation when it has the same name. Instead, I query just for instances with the same name (not all instances like before) and keep generating new names until we get one that isn't found in the system. Also added in the key pair deleting (this can be done without issue, see code). Let me know if you have any further thoughts!